packages/app/obojobo-document-engine/__tests__/oboeditor/util/keydown-util.test.js

```diff
@@ -348,11 +352,53 @@ describe('KeyDown Util', () => {
 
 		// make sure the inserted node is correct type
 		const insertedNode = Transforms.insertNodes.mock.calls[0][1]
-		expect(insertedNode.children[0]).toHaveProperty('type', 'ObojoboDraft.Chunks.Text')
-		expect(insertedNode.children[0]).toHaveProperty('subtype', 'ObojoboDraft.Chunks.Text.TextLine')
+		expect(insertedNode.children[0]).toHaveProperty('type', TEXT_NODE)
+		expect(insertedNode.children[0]).toHaveProperty('subtype', TEXT_LINE_NODE)
 		expect(insertedNode.children[0].children[0]).toEqual({ text: '' })
 	})
 
+	test('breakToText splits text', () => {
+		jest.spyOn(Transforms, 'insertNodes').mockReturnValue(true)
+		jest.spyOn(Node, 'fragment')
+
+		const editor = {
+			children: [
+				{
+					type: HEADING_NODE,
+					children: [{ text: 'someText' }]
+				}
+			],
+			selection: {
+				anchor: { path: [0, 0], offset: 4 },
+				focus: { path: [0, 0], offset: 4 }
+			},
+			isInline: () => false,
+			isVoid: () => false
+		}
+		ReactEditor.findPath.mockReturnValueOnce([0])
+
+		const event = {
+			isDefaultPrevented: () => false,
+			preventDefault: jest.fn()
+		}
+
+		KeyDownUtil.breakToText(event, editor, [editor.children[0], [0]], true)
+
+		expect(event.preventDefault).toHaveBeenCalled()
+		expect(Transforms.insertNodes).toHaveBeenCalled()
+
+		// make sure the correct node fragment is taken
+		const fragmentNode = Node.fragment.mock.results[0].value[0]
+		expect(fragmentNode).toHaveProperty('type', HEADING_NODE)
+		expect(fragmentNode.children[0]).toEqual({ text: 'Text' })
```

Comment:
Should this have the text of `some` since it split `someText`?

Reply:
The fragment is the part of the node that's being broken off, so it'd have `Text` in this case. The inserted node uses the text from the fragment node, so this is just confirming it has the correct source type and text.